tests/rptest/services/admin.py

```diff
@@ -236,34 +236,38 @@ def get_partition_leader(self, *, namespace, topic, partition):
     def transfer_leadership_to(self, *, namespace, topic, partition, target):
         """
         Looks up current ntp leader and transfer leadership to target node, 
-        this operations is NOP when current leader is the same as target. 
+        this operations is NOP when current leader is the same as target.
+        If user pass None for target this function will choose next replica for new leader.
         If leadership transfer was performed this function return True
         """
-        target_id = self.redpanda.idx(target) if isinstance(
-            target, ClusterNode) else target
+        def has_leader():
+            return self.get_partition_leader(
+                namespace=namespace, topic=topic, partition=partition) != -1
 
         #  check which node is current leader
-
-        def _get_details():
-            p = self.get_partitions(topic=topic,
-                                    partition=partition,
-                                    namespace=namespace)
-            self.redpanda.logger.debug(
-                f"ntp {namespace}/{topic}/{partition} details: {p}")
-            return p
-
-        def _has_leader():
-            return _get_details()['leader_id'] != -1
-
-        wait_until(_has_leader,
+        wait_until(has_leader,
                    timeout_sec=30,
                    backoff_sec=2,
                    err_msg="Failed to establish current leader")
 
-        details = _get_details()
-        if details['leader_id'] == target_id:
-            return False
-        path = f"raft/{details['raft_group_id']}/transfer_leadership?target={target_id}"
+        details = self.get_partitions(topic=topic,
+                                      partition=partition,
+                                      namespace=namespace)
+
+        self.redpanda.logger.debug(
+            f"ntp {namespace}/{topic}/{partition} details: {details}")
+
+        path = f"raft/{details['raft_group_id']}/transfer_leadership"
+
+        if target != None:
+            target_id = self.redpanda.idx(target) if isinstance(
+                target, ClusterNode) else target
+
+            if details['leader_id'] == target_id:
+                return False
+
+            path = f"{path}?target={target_id}"
```

Comment:
It looks to me like this patch could be much simpler using only the path manipulation based on `target` being provided or not is all that is needed.

It all looks good to me, I'm just wondering if I am missing something here in regards to the refactoring above (e.g. _has_leader -> has_leader) which seems to be unnecessary.